phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -4977,25 +4994,27 @@ public void removeConnection(PhoenixConnection connection) throws SQLException {
         if (returnSequenceValues) {
             ConcurrentMap<SequenceKey,Sequence> formerSequenceMap = null;
             synchronized (connectionCountLock) {
-                if (--connectionCount <= 0) {
-                    if (!this.sequenceMap.isEmpty()) {
-                        formerSequenceMap = this.sequenceMap;
-                        this.sequenceMap = Maps.newConcurrentMap();
+                if(!connection.isInternalConnection()) {
```

Comment:
We used to decrement `connectionCount` here before. Just commenting to make sure we are still  decrementing correctly.

Reply:
We decrement in the next block.